### DISCUSSION TITLE
How to extract vector images from page

### DISCUSSION THREAD

**Discussion by LastMonody** [pymupdf/PyMuPDF#1449] [Q&A]
Sorry, the first time I submitted a question, the wrong question became a bug. First, my purpose is to pass the page Getimagelist (full = true) determines whether there are pictures on the page. If there are pictures, I will parse them through OCR. If not, I will parse them through Fitz. But now the hand I call this method is empty, so I use Fitz to extract text, but it is empty. I also try two modes: vector image and bitmap, but bitmap can extract images, However, the page is a vector image, but the image cannot be obtained.

**Comment by JorjMcKie** [comment]
Ok, I  understand better now. BTW you could have continued comments with your first post - I converted it to a discussions already.

> If there are pictures, I will parse them through OCR. If not, I will parse them through Fitz.

1. You can now (since v1.19.0) OCR a page inside PyMuPDF / fitz. There is no need to use a diferent tool - if you are OK with using Tesseract for the OCR part.
2. I assume, by "vector image" you mean a PDF drawing? There are ways to extract this as well.
    - You can extract the drawings via `Page.get_drawings()` and reproduce them on a new PDF page.
    - You can simply make an image of the (full or partial) PDF page via `Page.get_pixmap(...)`.
3. You can use PyMuPDF's OCR feature to make a **_partial OCR_**: you can use automatic detection, which parts of the page are covered by images (which will then be OCRed) and normal text extraction will be used for the rest.

But if you just want images either drawings or raster, then you can simply check `page.get_images()` versus `page.get_drawings()`.
